azurerm/resource_arm_dns_zone_test.go

```diff
@@ -68,42 +68,47 @@ func testCheckAzureRMDnsZoneExists(name string) resource.TestCheckFunc {
 			return fmt.Errorf("Not found: %s", name)
 		}
 
-		conn := testAccProvider.Meta().(*ArmClient).rivieraClient
-
-		readRequest := conn.NewRequestForURI(rs.Primary.ID)
-		readRequest.Command = &dns.GetDNSZone{}
+		zoneName := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for DNS zone: %s", zoneName)
+		}
 
-		readResponse, err := readRequest.Execute()
+		conn := testAccProvider.Meta().(*ArmClient).zonesClient
+		resp, err := conn.Get(resourceGroup, zoneName)
 		if err != nil {
-			return fmt.Errorf("Bad: GetDNSZone: %s", err)
+			return fmt.Errorf("Bad: Get zone: %v", err)
 		}
-		if !readResponse.IsSuccessful() {
-			return fmt.Errorf("Bad: GetDNSZone: %s", readResponse.Error)
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: DNS zone %s (resource group: %s) does not exist", zoneName, resourceGroup)
 		}
 
 		return nil
 	}
 }
 
 func testCheckAzureRMDnsZoneDestroy(s *terraform.State) error {
-	conn := testAccProvider.Meta().(*ArmClient).rivieraClient
+	conn := testAccProvider.Meta().(*ArmClient).zonesClient
 
 	for _, rs := range s.RootModule().Resources {
-		if rs.Type != "azurerm_dns_zone" {
+		if rs.Type != "azurerm_dns_zone_record" {
 			continue
 		}
 
-		readRequest := conn.NewRequestForURI(rs.Primary.ID)
-		readRequest.Command = &dns.GetDNSZone{}
+		zoneName := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(resourceGroup, zoneName)
 
-		readResponse, err := readRequest.Execute()
 		if err != nil {
-			return fmt.Errorf("Bad: GetDNSZone: %s", err)
+			return nil
```

Comment:
when I added the error return, it failed 100% of the time. which I guess you could see coming since you suggested checking the 404 first. I really think we need to take this up with the team. I checked a bit and this is the pattern we follow in that routine.